File: c/calculus/matrix.c
#include <stdio.h>
#include <stdlib.h>

#define N 500 

void multiply(double a[N][N], double b[N][N], double result[N][N]) {
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            result[i][j] = 0.0;
            for (int k = 0; k < N; k++) {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }
}

int main() {
    double a[N][N]; 
    double b[N][N];
    double result[N][N];
    
    return 0;
}

File: fortran/calculus/matrix.f90
program multiply
    implicit none
    integer, parameter :: N = 500
    real*8 ::  result(N, N)
    

    integer :: i, j, k
    do i = 1, N
        do j = 1, N
            result(i, j) = 0.0d0
            do k = 1, N
                result(i, j) = result(i, j) + matrix(i, k) * matrix(k, j)
            end do
        end do
    end do
end program multiply